packages/env/lib/commands/exec.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Internal dependencies
+ */
+const initConfig = require( '../init-config' );
+const spawnDockerComposeRunCommand = require( '../spawn-docker-compose-run-command' );
+
+/**
+ * @typedef {import('../config').Config} Config
+ */
+
+/**
+ * Runs a script in the given environment.
+ *
+ * @param {Object}   options
+ * @param {string}   options.environment The environment to run the script in.
+ * @param {string[]} options.script      The script to run.
+ * @param {string[]} options.scriptArgs  The arguments to pass to the script.
+ * @param {boolean}  options.debug       True if debug mode is enabled.
+ * @param {Object}   options.spinner     A CLI spinner which indicates progress.
+ */
+module.exports = async function run( {
+	environment,
+	script,
+	scriptArgs,
+	spinner,
+	debug,
+} ) {
+	const config = await initConfig( { spinner, debug } );
+
+	// We want the script from the correct environment so that we can execute it.
+	const envConfig = config.env[ environment ];
```

Comment:
This was something that I had considered as well, but, I decided against it when I reviewed how `wp-env`'s configuration works. The `.wp-env.json` file describes how the environment should be configured, with support for overriding those options at the environment level. It seems like an outlier that this would be the only configuration option of this kind. Additionally, defining the environment it executes in when there's already an `"env"` key seems redundant.

If this is a sticking point, one possibility could be to support defaulting to an environment if only one has a script. So you could put `phpunit` in the `env.tests` option set and when you run `wp-env exec phpunit` it tries to find an environment to run it in. If there's more than one, it requires an option flag, otherwise, it executes it. I'm not really a fan of these kinds of opinions, however, since it feels like it wouldn't be clear what the command is actually doing.